### PR TITLE
PIM-10911: Add user-agent when sending an event

### PR DESCRIPTION
**Pull request by hogren**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: src/Akeneo/Connectivity/Connection/back/Infrastructure/Webhook/Client/GuzzleWebhookClient.php [hunk 1]
```diff
@@ -69,10 +55,16 @@ public function bulkSend(iterable $webhookRequests): void
                 $timestamp = \time();
                 $signature = Signature::createSignature($webhookRequest->secret(), $timestamp, $body);
 
+                $userAgent = 'AkeneoPIM/' . $this->versionProvider->getVersion();
+                if ($this->pfid) {
+                    $userAgent .= ' '.$this->pfid;
```
Comment (hogren) [inline src/Akeneo/Connectivity/Connection/back/Infrastructure/Webhook/Client/GuzzleWebhookClient.php:60]: IMO, it's preferable to use an explicit test e.g. `null !== $this->pfid`

File: src/Akeneo/Connectivity/Connection/back/Infrastructure/Webhook/Service/WebhookReachabilityChecker.php [hunk 2]
```diff
@@ -57,11 +56,16 @@ public function check(string $url, string $secret): UrlReachabilityStatus
 
         $timestamp = \time();
         $signature = Signature::createSignature($secret, $timestamp);
+        $userAgent = 'AkeneoPIM/' . $this->versionProvider->getVersion();
+        if ($this->pfid) {
+            $userAgent .= ' '.$this->pfid;
```
Comment (hogren) [inline src/Akeneo/Connectivity/Connection/back/Infrastructure/Webhook/Service/WebhookReachabilityChecker.php:61]: idem

File: src/Akeneo/Connectivity/Connection/back/tests/Unit/spec/Infrastructure/Webhook/Client/GuzzleWebhookClientSpec.php [hunk 3]
```diff
@@ -136,6 +145,13 @@ public function it_sends_webhook_requests_in_bulk(
         $signature = Signature::createSignature('a_secret', $timestamp, $body);
         Assert::assertEquals($signature, $request->getHeader(RequestHeaders::HEADER_REQUEST_SIGNATURE)[0]);
 
+        $userAgent = 'AkeneoPIM/v20210526040645';
+        if(getenv('PFID')) {
+            $userAgent .= ' '.getenv('PFID');
```
Comment (hogren) [inline src/Akeneo/Connectivity/Connection/back/tests/Unit/spec/Infrastructure/Webhook/Client/GuzzleWebhookClientSpec.php:150]: idem

**Review by eoliveauakeneo — APPROVED** [review]

**Review by masacc — APPROVED** [review]

**Comment by RiffVentura** [comment]
Don't forget to update the CHANGELOG :wink:

**Review by guillaumefer-akeneo — APPROVED** [review]

**Pull request merged**